### ISSUE TITLE
Unit tests for learning rate schedules and batch normalization

### ISSUE THREAD

**Issue by Ebanflo42** [BradenEverson/unda#92]
Write them lol

**Comment by Ebanflo42** [comment]
Also add better errors, no `assert` nonsense
